eth_utils/address.py

```diff
@@ -16,7 +18,10 @@
 )
 
 
-def is_hex_address(value):
+Address = NewType('Address', str)
```

Comment:
Hrm, we probably want two or even three of these.

```python
Address = NewType('Address', bytes)  # for canonical addresses
HexAddress = NewType('HexAddress', str)  # for hex encoded addresses
ChecksumAddress = NewType('ChecksumAddress', Address)  # for hex addresses with checksums.
```

Reply:
@pipermerriam going off this . . . in #100 I changed these defs to @davesque 's suggestion. the reason being that mypy would complain . . . for example. `to_canonical_address()` uses returns the result from `decode_hex()`, but `to_canonical_address()` expects and `Address` as it's return and not `bytes` which is what `decode_hex()` returns. Is that fine? It felt wrong changing `decode_hex()` to explicitly return an `Address` type

Reply:
For situations like that you can do `return Address(decode_hex(value))` to coerce the `bytes` return type to our custom `NewType` type.

Reply:
@pipermerriam Ahhh, cool - got it. One more question. As it currently stands, does this make sense?
```
Address = NewType('Address', bytes)  # for canonical addresses
HexAddress = NewType('HexAddress', str)  # for hex encoded addresses
ChecksumAddress = NewType('ChecksumAddress', Address)  # for hex addresses with checksums
```
Shouldn't `ChecksumAddress` be 
`ChecksumAddress = NewType('ChecksumAddress', HexAddress)`?